install/kubernetes/ingress-sni-zvpn.yaml

```diff
@@ -0,0 +1,364 @@
+              # listener_manager_impl.cc - if false, will attempt to convert to proto - not implemented for tcp_proxy
+              deprecated_v1: true
+              value:
+                stat_prefix: tcp_ca
+                route_config:
+                    routes:
+                    - cluster: zvpn.ca
+
+
+      clusters:
+      - name: rds
+        connect_timeout: 0.25s
+        type: STRICT_DNS
+        lb_policy: ROUND_ROBIN
+        hosts:
+          # TODO(costin): add TLS
+          - { socket_address: { address: "istio-pilot.istio-system", port_value: 8080 }}
+          # Local debugging/testing
+          #- { socket_address: { address: "localhost", port_value: 15008 }}
+
+      # Direct zvpn connection to pilot TLS port. This is used with a TCP proxy !
+      - name: zvpn.pilot
+        connect_timeout: 0.25s
+        type: STRICT_DNS
+        lb_policy: ROUND_ROBIN
+        hosts:
+          - { socket_address: { address: "istio-pilot", port_value: 15003 }}
+
+      # Connection to Istio CA. Service name must match the one for CA exposing gRPC.
+      - name: zvpn.ca
+        connect_timeout: 0.25s
+        lb_policy: ROUND_ROBIN
+        type: EDS
+        eds_cluster_config:
+          service_name: istio-ca-ilb.istio-system.svc.cluster.local|grpc
+          eds_config:
+            api_config_source:
+              api_type: REST_LEGACY
+              cluster_names: [rds]
+              refresh_delay: 10s
+
+      # Example/temp zvpn pointing to pilot, using its normal name, to verify we can create the tunnel.
+      # The service name should match what CDS generates - this section attempts to test merging of manual and CDS.
+      - name: out.pilot
+        connect_timeout: 0.25s
+        lb_policy: ROUND_ROBIN
+        type: EDS
+        eds_cluster_config:
+          service_name: istio-pilot.istio-system.svc.cluster.local|http-discovery
+          eds_config:
+            api_config_source:
+              api_type: REST_LEGACY
+              cluster_names: [rds]
+              refresh_delay: 10s
+
+
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-ingress
+  namespace: istio-system
+  annotations:
+    sidecar.istio.io/inject: "false"
+    keel.sh/pollSchedule: "@every 10m"
+  labels:
+    # Will auto-update on gcr.io pushes
+    keel.sh/policy: force
+    keel.sh/trigger: poll
+
+spec:
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+        keel.sh/pollSchedule: "@every 30m"
+      labels:
+        istio: ingress
+        # Will auto-update on gcr.io pushes
+        keel.sh/policy: force
+        keel.sh/trigger: poll
+    spec:
+      serviceAccountName: istio-ingress-service-account
+      containers:
+      - name: istio-ingress
+        # Manually start envoy, without using pilot-agent, so we can override the config.
+        command: ["/usr/local/bin/envoy"]
+        image: gcr.io/istio-testing/proxy_debug:costin
```

Comment:
Why? Use lyft/Envoy

Reply:
That's istio proxy - current build is flip-floping to the 3-month old envoy which doesn't work.
Will be obviously replaced with the template version. 

Also missing are the mixer filters...